src/networks.js

```diff
@@ -170,6 +172,12 @@ export const switchNetwork = async (chainId) => {
   }
 }
 
+export function isSupportedConnectedChain() {
+  if (window?.ethereum === undefined) return true
```

Comment:
It not should return `undefined` instead `true`?

Looks like the `isSupportedConnectedChain` check and should return `true` if the current chain is connected, right?

Addind `boolean | undefined` open the possibility of threat that behaviour, where the window.ethereum its not loaded correctly, maybe in mobile or browser without metamask, make sense?

I did not test nothing there, just looking in the code and trying understand the behaviou.

What you think?

Reply:
I need to test this further. But what I was going for it to be careful with mobile that doesn't have access to the window. In that case we don't show the wrong network button, but we keep the enable account one.

Reply:
If the function return false then the wrong networks button appears.

Reply:
I see, and if you return `undefined` instead and threat it like some error and use the default buttons?

Reply:
No. Its has 2 states, if true then its not a supported network, else it's supported. It's simple.

Reply:
No Felipe. The function just say if you are in the wrong network ( the chainid is not one of the supported networks ). Thats all. Maybe the name is not the best.